SmallBetterLogger/SmallBetterLogger.hpp

```diff
@@ -0,0 +1,179 @@
+		template<typename ...T>
+		inline void WriteLine(const std::string& message, const T& ...t);
+
+		// Writes to the stream a message and inserts values into placeholders (should they exist) and finishes with the newline character
+		template<typename ...T>
+		inline void WriteLine(const std::string&& message, const T&& ...t);
+
+	private:
+		// Private members
+		STREAM_TYPE m_StreamType;
+		std::fstream m_FileStream;
+		bool m_AutoFlush;
+
+		// Private methods
+
+		// Converts a T value to a string to be used in writing a log
+		template<typename T>
+		inline std::string stringConvert(const T& t);
+
+		// Replace all "{n}" placeholders with their respective values (n=0,...)
+		inline std::string replacePlaceholders(std::string message, std::vector<std::string>& items);
+
+		// Writes string to appropriate stream based on instance STREAM_TYPE (m_StreamType)
+		inline void writeToStream(const std::string&& str);
+	};
+
+	template<typename ...T>
+	inline void Logger::Write(const std::string& message, const T& ...t)
+	{
+		std::vector<std::string> printValues { stringConvert(t)... };
+		writeToStream(replacePlaceholders(message, printValues));
+	}
+
+	template<typename ...T>
+	inline void Logger::Write(const std::string&& message, const T&& ...t)
+	{
+		std::vector<std::string> printValues{ stringConvert(t)... };
+		writeToStream(replacePlaceholders(message, printValues));
+	}
+
+	template<typename ...T>
+	inline void Logger::WriteLine(const std::string& message, const T& ...t)
+	{
+		std::vector<std::string> printValues{ stringConvert(t)... };
+		writeToStream(replacePlaceholders(message, printValues) + "\n");
+	}
+
+	template<typename ...T>
+	inline void Logger::WriteLine(const std::string&& message, const T&& ...t)
+	{
+		std::vector<std::string> printValues{ stringConvert(t)... };
+		writeToStream(replacePlaceholders(message, printValues) + "\n");
+	}
+
+	template<typename T>
+	inline std::string Logger::stringConvert(const T& t)
+	{
+		std::stringstream ss;
+		ss << t;
+		return ss.str();
+	}
+
+	inline std::string sbl::Logger::replacePlaceholders(std::string message, std::vector<std::string>& items)
+	{
+		std::regex placeholder;
+
+		for (int i = 0; i < items.size(); i++)
+		{
+			placeholder = R"(\{)" + std::to_string(i) + R"(\})";
+			message = std::regex_replace(message, placeholder, items[i]);
+		}
+
+		return message;
+	}
+
+	inline void Logger::writeToStream(const std::string&& str)
+	{
+		switch (m_StreamType)
+		{
+		case STREAM_TYPE::STDERR:   std::cerr << str;       if (m_AutoFlush) std::cerr.flush();       break;
+		case STREAM_TYPE::STDLOG:   std::clog << str;       if (m_AutoFlush) std::clog.flush();       break;
+		case STREAM_TYPE::FILE:     m_FileStream << str;    if (m_AutoFlush) m_FileStream.flush();    break;
+		default:                    std::cout << str;       if (m_AutoFlush) std::cout.flush();       break;
```

Comment:
Even though the lines are small, you should still user the switch-case indentation convention, because you might encounter a section in your project where you need to add indentation and you lose consistency.